workbaskets/models.py

```diff
@@ -557,9 +558,15 @@ def new_transaction(self, **kwargs):
             )
 
         if "composite_key" not in kwargs:
-            kwargs["composite_key"] = (
-                f"{self.pk}-{kwargs['order']}-{kwargs['partition']}"
-            )
+            # Forms a composite key of a zero-padded 5 digit workbasket id + a randomly generated hexadecimal 11 character string.
+            # The probability of a collision is miniscule. It becomes 1% around 600,000 transactions and likely (>50%) at around 5 million.
+            workbasket_pk = str(self.pk).zfill(5)
+            if len(workbasket_pk) > 5:
+                raise ValueError(
+                    "Workbasket PK cannot be bigger than 5 digits for composite_key generation.",
+                )
```

Comment:
It might read better to directly apply the length test against `str(self.pk)` as it's its length that we're really interesting in checking.
```suggestion
            if len(str(self.pk)) > 5:
                # Take warning or error action.
            workbasket_pk = str(self.pk).zfill(5)
```